illumos-utils/src/svc.rs

```diff
@@ -47,6 +60,22 @@ mod inner {
                         == &smf::PropertyValue::Astring("online".to_string())
                     {
                         return Ok(());
+                    } else {
+                        if let Some(zname) = zone {
+                            if let Err(out) =
+                                tokio::process::Command::new(PFEXEC)
```

Comment:
I've put this here because services do not come up reliably when Omicron is running in a VM. I'm not sure if this is something we want in the mainline code to deal with early zone service failure or not. AFAICT this is only being called when waiting on zones to come to `milestone/single-user`. @smklein @andrewjstone @davepacheco.

Reply:
My gut is this is definitely _not_ something we want to be doing under normal operation.  If things are going into maintenance, I'd expect we need to figure out why and fix the underlying issues.  Without knowing what those are, it's hard to know we're not papering over a more significant issue or one that could come up again later.

Reply:
I've put this behind a compilation flag.

Reply:
Do we use that flag anywhere or is that just for development environments where people happen to be seeing this behavior?  Are you seeing this problem (services going into maintenance during startup) frequently?

I'm a little worried because to my knowledge we have not seen this before.  Might this change cause us to see this problem elsewhere?

Reply:
The flag is a one-off that one has to explicitly set for this code to be present. The goal is to make it hard to accidentally enable this.

I'm working in a [multi-switch falcon topology](https://github.com/oxidecomputer/testbed/tree/a4x2-topology/a4x2) that hosts the control plane. This topology is composed of propolis VMs interconnected by simnet links. In this environment I get many services going into maintenance mode every time I launch the control plane. The number and frequency of services going into maintenance on startup is high enough to make it very time-consuming to find and clear them all by hand. With this little workaround, things mostly come up automatically as the issues all appear to be transient in nature.

> I'm a little worried because to my knowledge we have not seen this before. Might this change cause us to see this problem elsewhere?

We started seeing this when I started running the control plane in topologies composed of VMs. Why this happens in VMs, I'm not sure - but I don't think anything here would cause this issue to manifest elsewhere. I think it's a bad interaction at the intersection of the control plane, SMF and running on virtual hardware.

Reply:
Thanks for that context!  That makes sense.  I think it's almost always a bug for any SMF service to go into maintenance on our systems.  If you haven't already, would you mind filing an issue for that (even if it's not high priority)?

Reply:
I filed #4399 on this.

Reply:
Thanks!